spoofax-pygments/spoofax_lexer.py

```diff
@@ -0,0 +1,126 @@
+import json
+import os
+import subprocess
+import sys
+import traceback
+
+import pygments.formatters
+from pygments.formatters.latex import LatexFormatter
+from pygments.lexer import Lexer
+from pygments.style import Style, StyleMeta
+from pygments.token import Token
+from pygments.util import add_metaclass
+
+
+# Subclass of Style that we can mess with without interfering with the original implementation
+class EmptyStyle(Style):
+    pass
+
+
+# This file is loaded when Pygmentize reads `-f spoofax_lexer.py` in its argument list.
+# The Lexer has two options:
+#     "language" - the language (SDF and ESV) to load from your Maven repository
+#     "parseTable" - an `sdf.tbl` file to use for parsing
+#     "esv" - an `editor.esv.af` file to use for coloring
+#     "preset" - the JSGLR2 parser preset to use (e.g. layoutSensitive)
+class CustomLexer(Lexer):
```

Comment:
Isn't `SpoofaxLexer` a more descriptive name?

Reply:
Yes, but as stated in [the Pygments docs](http://pygments.org/docs/cmdline/#custom-lexers-and-formatters):

> The `-x` flag enables custom lexers and formatters to be loaded from files relative to the current directory. Create a file with a class named CustomLexer or CustomFormatter, then specify it on the command line:
>
>     $ pygmentize -l your_lexer.py -f your_formatter.py -x
> You can also specify the name of your class with a colon:
>
>     $ pygmentize -l your_lexer.py:SomeLexer -x

By naming the class `CustomLexer`, it saves having to write `spoofax_lexer.py:SpoofaxLexer` in the Minted config, because this can now be shortened to `spoofax_lexer.py`. The lexer is the same for all Spoofax languages now anyway :slightly_smiling_face:

Reply:
Ah yes, I was wondering why you were able to leave it out